packages/data/src/plugins/controls/index.js

```diff
@@ -22,9 +23,24 @@ export default function( registry ) {
 					store,
 					enhancer( createStore )( options.reducer )
 				);
+
+				registry.namespaces[ reducerKey ].supportControls = true;
 			}
 
 			return store;
 		},
+
+		async fulfill( reducerKey, selectorName, ...args ) {
+			if ( ! registry.namespaces[ reducerKey ].supportControls ) {
+				registry.fulfill( reducerKey, selectorName, ...args );
+				return;
+			}
+
+			const resolver = get( registry.namespaces, [ reducerKey, 'resolvers', selectorName ] );
```

Comment:
I was under the impression that part of the intent with introducing the promise middleware was to make it so that at the very least, `controls` doesn't need to know of resolvers, and at best `controls` doesn't even need to implement `fulfill` at all.

Why again did we add the promise middleware? Just to refactor async generators?

Reply:
I'm pretty sure I explained this elsewhere :)

The promise middleware is a way to support resolvers without any plugins:

 - A resolver is action creator returning a promise

And yes, we don't need `fulfill` to implement `controls` support for resolvers BUT we do need it temporarily to support "async-generators" because async generators (like we write them today) can't be considered action creators because they take "state" as first argument, which means I needed a way to override the default `fulfill` for async generators. And by doing so, I needed to restore the default `fulfill` behavior for controls.

In other words:

 - It's temporary
 - When we remove the async-generators plugin, we will remove entirely the `fulfill` and `namespaces` from the registry object and we will also remove the `fulfill` defined in the controls plugin.

Reply:
Should we be deprecating `async-generator` then? Now? In a follow-up pull request?

Reply:
yes 

> I wanted to deprecated the async-generator middleware/plugin but thought I would first wait for us to refactor the current async generators we have in Gutenberg to use controls and do that right after.

Reply:
>we will remove entirely the fulfill and namespaces from the registry object and we will also remove the fulfill defined in the controls plugin.

Should we mark as experimental then, to avoid need for deprecation?